File: diag/blinkenLEDTest.py
import RPi.GPIO as GPIO
import time, random

LEDdPIN = 36
LEDlPIN = 38
LEDcPIN = 40
SWdPIN = 35
SWlPIN = 33
SWcPIN = 37
MSBFIRST = 1

GPIO.setmode(GPIO.BOARD)
GPIO.setup((LEDdPIN,LEDlPIN,LEDcPIN,SWdPIN, SWcPIN, SWlPIN), GPIO.OUT)
GPIO.setup(SWdPIN,GPIO.IN)

#shiftOut(LdataPin,LclockPin,MSBFIRST,bus.state >> 8);
def shiftOut(dpin,cpin,order,idata):
  #MSBOrder is 1 always for us
  for i in range(7, -1, -1):
    GPIO.output(cpin,False)
    GPIO.output(dpin, (idata >> i) & 0x01 )    
    GPIO.output(cpin,True)
  return

while (True):
  busdata = random.randrange(0xFF)
  busaddress = random.randrange(0xFFFF)
  busstate = random.randrange(0xFFFF)
  GPIO.output(LEDlPIN ,False)
  shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,busstate >> 8)
  shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,busaddress >> 8)
  shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,busaddress)
  shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,busstate)
  shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,busdata)
  GPIO.output(LEDlPIN ,True)
  time.sleep(.5)

GPIO.cleanup()

File: diag/blinkenLEDTest.c
#include <pigpio.h>

#define LEDdPIN  16
#define LEDlPIN  20
#define LEDcPIN  21
#define SWdPIN  19
#define SWlPIN  13
#define SWcPIN  26
#define MSBFIRST  1

void shiftOut(uint8_t dpin,uint8_t cpin,uint8_t order,uint8_t idata) {
  //MSBOrder is 1 always for us
  for(int i=7; i > -1; i--){
    gpioWrite(cpin,PI_LOW);
    gpioWrite(dpin, (idata >> i) & 0x01 );
    gpioWrite(cpin,PI_HIGH);
  }
}

void rpi_init()
{
    gpioInitialise();     //Setup the library
    gpioSetMode(LEDlPIN,PI_OUTPUT);
    gpioSetMode(LEDlPIN,PI_OUTPUT);
    gpioSetMode(LEDcPIN, PI_OUTPUT);
    gpioSetMode(SWdPIN, PI_INPUT);
    gpioSetMode(SWcPIN, PI_INPUT);
    gpioSetMode(SWlPIN, PI_OUTPUT);

}

int main ()
{
    rpi_init();
    uint16_t status = 0xA5A5;
    uint8_t data = 0xA5;
    uint16_t bus = 0xA5A5;
    //uint16_t *bus_switches input from panel
    //uint16_t *cmd_switches input from panel

    while (1) {
       gpioWrite(LEDlPIN, PI_LOW);
       // Now push data to 74HC595
       shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,status >> 8);
       shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,bus >> 8);
       shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,bus);
       shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,status);
       shiftOut(LEDdPIN,LEDcPIN,MSBFIRST,data);
       // take the latch pin high so the LEDs will light up:
       gpioWrite(LEDlPIN, PI_HIGH);
       gpioDelay(500000);
      status = rand() % 65536;
      data = rand() % 256;
      bus = rand() % 65536;
    }

}